### PR TITLE
Package refactor

### PR DESCRIPTION
**Pull request by SamYuan1990**
#### Type of change

- Bug fix
- New feature
- Improvement (improvement to code, performance, etc)
- Test update
- Documentation update

### CODE REVIEW

File: bccsp/factory/factory.go [hunk 1]
```diff
@@ -55,6 +55,16 @@ func GetDefault() bccsp.BCCSP {
 }
 
 func initBCCSP(f BCCSPFactory, config *FactoryOpts) (bccsp.BCCSP, error) {
+	// global var init
```
Comment (davidkhala) [inline bccsp/factory/factory.go]: [code hygience]

Comment (SamYuan1990) [reply bccsp/factory/factory.go]: yes, it is tbd code parts. @guijunchen , could you please help adding sm2 logic in this code part?

File: bccsp/idemix/handlers/revocation.go [hunk 2]
```diff
@@ -83,7 +83,8 @@ func NewRevocationPublicKey(pubKey *ecdsa.PublicKey) *revocationPublicKey {
 // Bytes converts this key to its byte representation,
 // if this operation is allowed.
 func (k *revocationPublicKey) Bytes() (raw []byte, err error) {
-	// TWGC todo
+	// TWGC won't do
```
Comment (davidkhala) [inline bccsp/idemix/handlers/revocation.go]: revert because WON'T DO

File: bccsp/sm2/sm2key.go [hunk 3]
```diff
@@ -15,10 +12,13 @@ limitations under the License.
...
 package sm2
 
-// TWGC todo
 import (
 	"crypto/elliptic"
+	"crypto/rand"
 	"crypto/sha256"
+	"encoding/asn1"
+	"encoding/pem"
+	"errors"
```
Comment (SamYuan1990) [inline bccsp/sm2/sm2key.go:21]: todo: 	"github.com/Hyperledger-TWGC/ccs-gm/x509"

**Comment by SamYuan1990** [comment]
@davidkhala , please help with this PR.

File: bccsp/sm2/sm2key.go [hunk 4]
```diff
@@ -127,3 +152,193 @@ func (k *SM2PublicKey) PublicKey() (bccsp.Key, error) {
...
+	}
+
+	return pem.EncodeToMemory(block), nil
+}
+
+func SM2PrivateKeyToDER(privateKey interface{}) ([]byte, error) {
+	return x509.MarshalECPrivateKey(privateKey.(*sm2.PrivateKey))
+}
+
+func SM2privateKeyToPEM(k interface{}, pwd []byte) ([]byte, error) {
```
Comment (davidkhala) [inline bccsp/sm2/sm2key.go:183]: 这两个为何差别那么大？

Comment (SamYuan1990) [reply bccsp/sm2/sm2key.go:183]: @guijunchen , 从原版粘过来的，不排除有bug。随着加test case慢慢捋顺吧。

Comment (SamYuan1990) [reply bccsp/sm2/sm2key.go:183]: > 这两个为何差别那么大？

https://github.com/Hyperledger-TWGC/fabric-gm-wiki/issues/34 最近我看sm3的相关代码感觉也有些存疑的地方。@davidkhala, 如果发现存疑的地方，一并列在这里边吧。

**Pull request merged**